TIOCGDEV is used to get the kernel device name behind /dev/console
Temoprary solution until another solution exists.

---
 usr/include/asm-arm/ioctls.h     |    1 +
 usr/include/asm-ia64/ioctls.h    |    1 +
 usr/include/asm-parisc/ioctls.h  |    1 +
 usr/include/asm-powerpc/ioctls.h |    1 +
 usr/include/asm-s390/ioctls.h    |    1 +
 usr/include/asm-sparc/ioctls.h   |    1 +
 usr/include/asm-sparc64/ioctls.h |    1 +
 usr/include/asm-x86/ioctls.h     |    1 +
 6 files changed, 6 insertions(+)

--- a/usr/include/asm-arm/ioctls.h
+++ b/usr/include/asm-arm/ioctls.h
@@ -52,6 +52,7 @@
 #define TCSETSF2	_IOW('T',0x2D, struct termios2)
 #define TIOCGPTN	_IOR('T',0x30, unsigned int) /* Get Pty Number (of pty-mux device) */
 #define TIOCSPTLCK	_IOW('T',0x31, int)  /* Lock/unlock Pty */
+#define TIOCGDEV	_IOR('T',0x32, unsigned int) /* Get real dev no below /dev/console */
 
 #define FIONCLEX	0x5450  /* these numbers need to be adjusted. */
 #define FIOCLEX		0x5451
--- a/usr/include/asm-ia64/ioctls.h
+++ b/usr/include/asm-ia64/ioctls.h
@@ -55,6 +55,7 @@
 #define TIOCGSID	0x5429  /* Return the session ID of FD */
 #define TIOCGPTN	_IOR('T',0x30, unsigned int) /* Get Pty Number (of pty-mux device) */
 #define TIOCSPTLCK	_IOW('T',0x31, int)  /* Lock/unlock Pty */
+#define TIOCGDEV	_IOR('T',0x32, unsigned int) /* Get real dev no below /dev/console */
 
 #define FIONCLEX	0x5450  /* these numbers need to be adjusted. */
 #define FIOCLEX		0x5451
--- a/usr/include/asm-parisc/ioctls.h
+++ b/usr/include/asm-parisc/ioctls.h
@@ -48,6 +48,7 @@
 #define TIOCGSID	_IOR('T', 20, int) /* Return the session ID of FD */
 #define TIOCGPTN	_IOR('T',0x30, unsigned int) /* Get Pty Number (of pty-mux device) */
 #define TIOCSPTLCK	_IOW('T',0x31, int)  /* Lock/unlock Pty */
+#define TIOCGDEV	_IOR('T',0x32, unsigned int) /* Get real dev no below /dev/console */
 
 #define FIONCLEX	0x5450  /* these numbers need to be adjusted. */
 #define FIOCLEX		0x5451
--- a/usr/include/asm-powerpc/ioctls.h
+++ b/usr/include/asm-powerpc/ioctls.h
@@ -91,6 +91,7 @@
 #define TIOCGSID	0x5429  /* Return the session ID of FD */
 #define TIOCGPTN	_IOR('T',0x30, unsigned int) /* Get Pty Number (of pty-mux device) */
 #define TIOCSPTLCK	_IOW('T',0x31, int)  /* Lock/unlock Pty */
+#define TIOCGDEV	_IOR('T',0x32, unsigned int) /* Get real dev no below /dev/console */
 
 #define TIOCSERCONFIG	0x5453
 #define TIOCSERGWILD	0x5454
--- a/usr/include/asm-s390/ioctls.h
+++ b/usr/include/asm-s390/ioctls.h
@@ -56,6 +56,7 @@
 #define TIOCGSID	0x5429  /* Return the session ID of FD */
 #define TIOCGPTN	_IOR('T',0x30, unsigned int) /* Get Pty Number (of pty-mux device) */
 #define TIOCSPTLCK	_IOW('T',0x31, int)  /* Lock/unlock Pty */
+#define TIOCGDEV	_IOR('T',0x32, unsigned int) /* Get real dev no below /dev/console */
 
 #define FIONCLEX	0x5450  /* these numbers need to be adjusted. */
 #define FIOCLEX		0x5451
--- a/usr/include/asm-sparc/ioctls.h	2007-07-08 19:32:17.000000000 -0400
+++ b/usr/include/asm-sparc/ioctls.h	2007-08-27 14:01:21.000000000 -0400
@@ -15,6 +15,7 @@
 #define TCSETS2		_IOW('T', 13, struct termios2)
 #define TCSETSW2	_IOW('T', 14, struct termios2)
 #define TCSETSF2	_IOW('T', 15, struct termios2)
+#define TIOCGDEV	_IOR('T',0x32, unsigned int) /* Get real dev no below /dev/console */
 
 /* Note that all the ioctls that are not available in Linux have a 
  * double underscore on the front to: a) avoid some programs to
--- a/usr/include/asm-sparc64/ioctls.h	2007-07-08 19:32:17.000000000 -0400
+++ b/usr/include/asm-sparc64/ioctls.h	2007-08-27 14:01:21.000000000 -0400
@@ -16,6 +16,7 @@
 #define TCSETS2		_IOW('T', 13, struct termios2)
 #define TCSETSW2	_IOW('T', 14, struct termios2)
 #define TCSETSF2	_IOW('T', 15, struct termios2)
+#define TIOCGDEV	_IOR('T',0x32, unsigned int) /* Get real dev no below /dev/console */
 
 /* Note that all the ioctls that are not available in Linux have a 
  * double underscore on the front to: a) avoid some programs to
--- a/usr/include/asm-x86/ioctls.h
+++ b/usr/include/asm-x86/ioctls.h
@@ -52,6 +52,7 @@
 #define TCSETSF2	_IOW('T',0x2D, struct termios2)
 #define TIOCGPTN	_IOR('T',0x30, unsigned int) /* Get Pty Number (of pty-mux device) */
 #define TIOCSPTLCK	_IOW('T',0x31, int)  /* Lock/unlock Pty */
+#define TIOCGDEV	_IOR('T',0x32, unsigned int) /* Get real dev no below /dev/console */
 
 #define FIONCLEX	0x5450  /* these numbers need to be adjusted. */
 #define FIOCLEX		0x5451
